src/condor_shadow.V6.1/pseudo_ops.cpp

```diff
@@ -941,7 +944,175 @@ pseudo_event_notification( const ClassAd & ad ) {
 		// exit-to-requeue routine here.
 		//
 		return 0;
+	} else if( eventType == ETYPE_DIAGNOSTIC_RESULT ) {
+		std::string diagnostic;
+		if(! ad.LookupString( ATTR_DIAGNOSTIC, diagnostic )) {
+			dprintf( D_ALWAYS, "Starter sent a diagnostic result, but did not name which diagnostic; ignoring.\n" );
+			return -1;
+		}
+
+        if( diagnostic != DIAGNOSTIC_SEND_EP_LOGS ) {
+            dprintf( D_ALWAYS, "Starter sent an unexpected diagnostic result (for '%s'); ignoring.\n", diagnostic.c_str() );
+            return -1;
+        }
+
+		std::string result;
+		if(! ad.LookupString( "Result", result ) ) {
+			dprintf( D_ALWAYS, "Starter sent a diagnostic result for '%s', but it had no result.\n", diagnostic.c_str() );
+			return -1;
+		}
+
+		if( result != "Completed" ) {
+			dprintf( D_ALWAYS, "Diagnostic '%s' did not complete: '%s'\n",
+				diagnostic.c_str(), result.c_str()
+			);
+			return 0;
+		}
+
+
+		int exitStatus;
+		if(! ad.LookupInteger( "ExitStatus", exitStatus ) ) {
+			dprintf( D_ALWAYS, "Starter sent a completed diagnostic result for '%s', but it had no exit status.\n", diagnostic.c_str() );
+			return -1;
+		}
+
+		if( exitStatus != 0 ) {
+			dprintf( D_ALWAYS, "Starter sent a completed diagnostic result for '%s', but its exit status was non-zero (%d)\n", diagnostic.c_str(), exitStatus );
+			return 0;
+		}
+
+
+		std::string contents;
+		if(! ad.LookupString( "Contents", contents ) ) {
+			dprintf( D_ALWAYS, "Starter sent a completed diagnostic result for '%s', but it had no contents.\n", diagnostic.c_str() );
+			return -1;
+		}
+
+		int decoded_bytes = 0;
+		unsigned char * decoded;
+		condor_base64_decode( contents.c_str(), & decoded, & decoded_bytes, false );
+		decoded[decoded_bytes] = '\0';
+
+		// Write `decoded` to a well-known location.  We should probably
+		// add a job-ad attribute which controls the location.
+		std::string jobIWD;
+		if( jobAd->LookupString( ATTR_JOB_IWD, jobIWD ) ) {
+			std::filesystem::path iwd(jobIWD);
+			std::filesystem::path diagnostic_dir( iwd / ".diagnostic" );
+
+			if(! std::filesystem::exists( diagnostic_dir )) {
+				std::error_code ec;
+				std::filesystem::create_directory( diagnostic_dir, ec );
+			}
```

Comment:
We carry on and print a nice error message, including the full path to the missing directory, if the file can't be created.  It would be a little clearer to note that it's because we tried and failed to create the directory, but we also don't complain if the path exists but isn't a directory.  I can add the additional error checks in, but this is going into the shadow log, not a hold message, so I thought the extra details weren't necessary.